patch-package
--- a/node_modules/react-native-screens/src/screens.js
+++ b/node_modules/react-native-screens/src/screens.js
@@ -5,6 +5,7 @@ import {
   View,
   UIManager,
   StyleSheet,
+  Platform,
 } from 'react-native';
 
 let USE_SCREENS = false;
@@ -22,14 +23,16 @@ export function screensEnabled() {
   return USE_SCREENS;
 }
 
-const NativeScreen = Animated.createAnimatedComponent(
+const isPlatformSupported = ['android', 'ios'].includes(Platform.OS);
+
+const NativeScreen = isPlatformSupported ? Animated.createAnimatedComponent(
   requireNativeComponent('RNSScreen', null)
-);
+) : null;
 
-const NativeScreenContainer = requireNativeComponent(
+const NativeScreenContainer = isPlatformSupported ? requireNativeComponent(
   'RNSScreenContainer',
   null
-);
+) : null;
 
 export class Screen extends React.Component {
   setNativeProps(props) {
